### PR TITLE
Make landing categories great again (with the ability to configure)

### PR DESCRIPTION
**Pull request by JamelleG**
This adds the ability to configure different landing category pages using a custom config file `layout_categories.xml` which is read and then injected into the same functionality that the module currently uses but this makes it driven by config values instead of hardcoded. (which was not a requirement at the time but now it is so yeah)

### CODE REVIEW

**Review by annybs — DISMISSED** [review]
LGTM 👌 Given that this has to do with https://github.com/WeareJH/trend-machinery/pull/51 I would recommend also tagging the appropriate version after merge. There's a reference to version 1.2.2 somewhere in your code, but check with the primary code owner of this module.

File: Plugin/Category/Attribute/Source/ModePlugin.php
Comment (dcabrejas) [inline Plugin/Category/Attribute/Source/ModePlugin.php]: Remove this constant. It's no longer being used.

File: Model/Widget/Instance.php
Comment (dcabrejas) [inline Model/Widget/Instance.php]: Remove this constant. It's no longer used.

File: Model/Widget/Instance.php
Comment (dcabrejas) [inline Model/Widget/Instance.php]: Could we import all those fully qualified names at the top of the file instead using `use` statements please?

File: Block/Adminhtml/Widget/Instance/Edit/Tab/Main/Layout.php
Comment (dcabrejas) [inline Block/Adminhtml/Widget/Instance/Edit/Tab/Main/Layout.php:78]: Can more than one category view type share the same label and code? Does this not create conflicts?

File: Block/Adminhtml/Widget/Instance/Edit/Tab/Main/Layout.php
Comment (dcabrejas) [inline Block/Adminhtml/Widget/Instance/Edit/Tab/Main/Layout.php]: Could we import this FQNs using `use` statements at the top of the file please?

**Review by dcabrejas — CHANGES_REQUESTED** [review]

File: Block/Adminhtml/Widget/Instance/Edit/Tab/Main/Layout.php
Comment (JamelleG) [reply Block/Adminhtml/Widget/Instance/Edit/Tab/Main/Layout.php:78]: In this case no because the code/label is applied to the group type instead of the specific view.
They also do the same for each product type (simple,configurable) all have the label 'Products'

**Review by dcabrejas — APPROVED** [review]

**Comment by dcabrejas** [comment]
Hi @JamelleG Good work! Let's get this merged and tagged.

**Pull request merged**
